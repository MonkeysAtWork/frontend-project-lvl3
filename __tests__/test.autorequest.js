import nock from 'nock';
import path from 'path';
import fs from 'fs';
import { html } from 'js-beautify';

import run from '../src/application';


nock.disableNetConnect();

const fixturesPath = path.join(__dirname, '__fixtures__');

beforeAll(() => {
  const pathToTemplate = path.join(__dirname, '..', 'template.html');
  const pathToIndex = path.join(fixturesPath, 'index.html');
  fs.copyFileSync(pathToTemplate, pathToIndex);
});

const response = fs.readFileSync(path.join(fixturesPath, 'rss_0.xml'), 'utf-8');
const response2 = fs.readFileSync(path.join(fixturesPath, 'rss_1.xml'), 'utf-8');
const response3 = fs.readFileSync(path.join(fixturesPath, 'rss_2.xml'), 'utf-8');

const delay = time => new Promise(resolve => setTimeout(resolve, time));
const getTree = () => html(document.body.innerHTML);
nock('https://cors-anywhere.herokuapp.com')
  // .log(console.log)
  .get(/auto$/)
  .twice()
  .reply(200, response)
  .get(/auto$/)
  .reply(200, response2)
  .get(/auto$/)
  .replyWithError('some error')
  .get(/auto$/)
  .reply(200, response3);


const initHtml = fs.readFileSync(path.join(fixturesPath, 'index.html'), 'utf-8');
document.documentElement.innerHTML = initHtml;
run();

const form = document.querySelector('#rssAddForm');
const input = form.elements.link;
const url = 'http://www.test.com';
const inputEvent = new Event('input', { bubbles: true });
const submitEvent = new Event('submit');


test('auto request', () => {
  input.value = `${url}/auto`;
  input.dispatchEvent(inputEvent);
  return delay(10)
    .then(() => form.dispatchEvent(submitEvent))
    .then(() => delay(200))
    .then(() => expect(getTree()).toMatchSnapshot())
    .then(() => delay(5200))
    .then(() => expect(getTree()).toMatchSnapshot())
    .then(() => delay(5200))
    .then(() => expect(getTree()).toMatchSnapshot())
    .then(() => delay(5200))
    .then(() => expect(getTree()).toMatchSnapshot())
    .then(() => delay(5200))
    .then(() => expect(getTree()).toMatchSnapshot());
}, 30000);
