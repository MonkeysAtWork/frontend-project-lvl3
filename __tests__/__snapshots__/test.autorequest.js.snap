// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auto request 1`] = `
"<div class=\\"jumbotron pt-2 pb-3 mb-0\\" id=\\"header\\">
    <h1 class=\\"display-5 \\">RSS aggregator</h1>
    <div id=\\"formConteiner\\">
        <form class=\\"form-inline\\" id=\\"rssAddForm\\">
            <input type=\\"text\\" name=\\"link\\" class=\\"form-control col mr-2\\" spellcheck=\\"false\\" placeholder=\\"RSS link\\">
            <button type=\\"submit\\" class=\\"btn btn btn-outline-primary d-flex align-items-center\\" disabled=\\"true\\">Add</button>
            <div class=\\"invalid-feedback\\" id=\\"feedback\\"></div>
        </form>
    </div>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div class=\\"col col-3 p-0\\" id=\\"feeds\\">
            <div class=\\"list-group list-group-flush border-right\\">
                <div class=\\"list-group-item\\">
                    <h5 class=\\"mb-1\\">Lorem ipsum feed for an interval of 1 minutes</h5>
                    <p class=\\"mb-1\\">This is a constantly updating lorem ipsum feed</p>
                </div>
            </div>
        </div>
        <div class=\\"col col-8\\" id=\\"posts\\">
            <ul class=\\"list-group list-group-flush overflow-auto\\">
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637680\\">Veniam magna occaecat qui irure mollit ea consequat ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637620\\">Do reprehenderit nostrud laborum deserunt aliqua aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637560\\">Qui amet et in pariatur et nostrud quis do id.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637500\\">Tempor magna non duis sint deserunt pariatur adipisicing elit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637440\\">Incididunt consequat reprehenderit sit adipisicing voluptate sint consequat.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637380\\">Labore aliqua incididunt duis voluptate reprehenderit officia est mollit voluptate velit consectetur nulla velit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637320\\">Voluptate deserunt elit consectetur tempor dolore ut dolor ex ullamco sunt aliquip esse sit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637260\\">Eu adipisicing nostrud tempor voluptate commodo laboris eiusmod voluptate ex.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637200\\">Sint anim reprehenderit elit commodo.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637140\\">Ea officia veniam dolore adipisicing nulla esse aliqua.</a></li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`auto request 2`] = `
"<div class=\\"jumbotron pt-2 pb-3 mb-0\\" id=\\"header\\">
    <h1 class=\\"display-5 \\">RSS aggregator</h1>
    <div id=\\"formConteiner\\">
        <form class=\\"form-inline\\" id=\\"rssAddForm\\">
            <input type=\\"text\\" name=\\"link\\" class=\\"form-control col mr-2\\" spellcheck=\\"false\\" placeholder=\\"RSS link\\">
            <button type=\\"submit\\" class=\\"btn btn btn-outline-primary d-flex align-items-center\\" disabled=\\"true\\">Add</button>
            <div class=\\"invalid-feedback\\" id=\\"feedback\\"></div>
        </form>
    </div>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div class=\\"col col-3 p-0\\" id=\\"feeds\\">
            <div class=\\"list-group list-group-flush border-right\\">
                <div class=\\"list-group-item\\">
                    <h5 class=\\"mb-1\\">Lorem ipsum feed for an interval of 1 minutes</h5>
                    <p class=\\"mb-1\\">This is a constantly updating lorem ipsum feed</p>
                </div>
            </div>
        </div>
        <div class=\\"col col-8\\" id=\\"posts\\">
            <ul class=\\"list-group list-group-flush overflow-auto\\">
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637680\\">Veniam magna occaecat qui irure mollit ea consequat ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637620\\">Do reprehenderit nostrud laborum deserunt aliqua aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637560\\">Qui amet et in pariatur et nostrud quis do id.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637500\\">Tempor magna non duis sint deserunt pariatur adipisicing elit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637440\\">Incididunt consequat reprehenderit sit adipisicing voluptate sint consequat.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637380\\">Labore aliqua incididunt duis voluptate reprehenderit officia est mollit voluptate velit consectetur nulla velit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637320\\">Voluptate deserunt elit consectetur tempor dolore ut dolor ex ullamco sunt aliquip esse sit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637260\\">Eu adipisicing nostrud tempor voluptate commodo laboris eiusmod voluptate ex.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637200\\">Sint anim reprehenderit elit commodo.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637140\\">Ea officia veniam dolore adipisicing nulla esse aliqua.</a></li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`auto request 3`] = `
"<div class=\\"jumbotron pt-2 pb-3 mb-0\\" id=\\"header\\">
    <h1 class=\\"display-5 \\">RSS aggregator</h1>
    <div id=\\"formConteiner\\">
        <form class=\\"form-inline\\" id=\\"rssAddForm\\">
            <input type=\\"text\\" name=\\"link\\" class=\\"form-control col mr-2\\" spellcheck=\\"false\\" placeholder=\\"RSS link\\">
            <button type=\\"submit\\" class=\\"btn btn btn-outline-primary d-flex align-items-center\\" disabled=\\"true\\">Add</button>
            <div class=\\"invalid-feedback\\" id=\\"feedback\\"></div>
        </form>
    </div>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div class=\\"col col-3 p-0\\" id=\\"feeds\\">
            <div class=\\"list-group list-group-flush border-right\\">
                <div class=\\"list-group-item\\">
                    <h5 class=\\"mb-1\\">Lorem ipsum feed for an interval of 1 minutes</h5>
                    <p class=\\"mb-1\\">This is a constantly updating lorem ipsum feed</p>
                </div>
            </div>
        </div>
        <div class=\\"col col-8\\" id=\\"posts\\">
            <ul class=\\"list-group list-group-flush overflow-auto\\">
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637800\\">Veniam magna qui mollit ea ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637740\\">Do nostrud laborum aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637680\\">Veniam magna occaecat qui irure mollit ea consequat ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637620\\">Do reprehenderit nostrud laborum deserunt aliqua aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637560\\">Qui amet et in pariatur et nostrud quis do id.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637500\\">Tempor magna non duis sint deserunt pariatur adipisicing elit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637440\\">Incididunt consequat reprehenderit sit adipisicing voluptate sint consequat.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637380\\">Labore aliqua incididunt duis voluptate reprehenderit officia est mollit voluptate velit consectetur nulla velit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637320\\">Voluptate deserunt elit consectetur tempor dolore ut dolor ex ullamco sunt aliquip esse sit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637260\\">Eu adipisicing nostrud tempor voluptate commodo laboris eiusmod voluptate ex.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637200\\">Sint anim reprehenderit elit commodo.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637140\\">Ea officia veniam dolore adipisicing nulla esse aliqua.</a></li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`auto request 4`] = `
"<div class=\\"jumbotron pt-2 pb-3 mb-0\\" id=\\"header\\">
    <h1 class=\\"display-5 \\">RSS aggregator</h1>
    <div id=\\"formConteiner\\">
        <form class=\\"form-inline\\" id=\\"rssAddForm\\">
            <input type=\\"text\\" name=\\"link\\" class=\\"form-control col mr-2\\" spellcheck=\\"false\\" placeholder=\\"RSS link\\">
            <button type=\\"submit\\" class=\\"btn btn btn-outline-primary d-flex align-items-center\\" disabled=\\"true\\">Add</button>
            <div class=\\"invalid-feedback\\" id=\\"feedback\\"></div>
        </form>
    </div>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div class=\\"col col-3 p-0\\" id=\\"feeds\\">
            <div class=\\"list-group list-group-flush border-right\\">
                <div class=\\"list-group-item\\">
                    <h5 class=\\"mb-1\\">Lorem ipsum feed for an interval of 1 minutes</h5>
                    <p class=\\"mb-1\\">This is a constantly updating lorem ipsum feed</p>
                </div>
            </div>
        </div>
        <div class=\\"col col-8\\" id=\\"posts\\">
            <ul class=\\"list-group list-group-flush overflow-auto\\">
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637800\\">Veniam magna qui mollit ea ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637740\\">Do nostrud laborum aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637680\\">Veniam magna occaecat qui irure mollit ea consequat ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637620\\">Do reprehenderit nostrud laborum deserunt aliqua aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637560\\">Qui amet et in pariatur et nostrud quis do id.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637500\\">Tempor magna non duis sint deserunt pariatur adipisicing elit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637440\\">Incididunt consequat reprehenderit sit adipisicing voluptate sint consequat.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637380\\">Labore aliqua incididunt duis voluptate reprehenderit officia est mollit voluptate velit consectetur nulla velit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637320\\">Voluptate deserunt elit consectetur tempor dolore ut dolor ex ullamco sunt aliquip esse sit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637260\\">Eu adipisicing nostrud tempor voluptate commodo laboris eiusmod voluptate ex.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637200\\">Sint anim reprehenderit elit commodo.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637140\\">Ea officia veniam dolore adipisicing nulla esse aliqua.</a></li>
            </ul>
        </div>
    </div>
</div>"
`;

exports[`auto request 5`] = `
"<div class=\\"jumbotron pt-2 pb-3 mb-0\\" id=\\"header\\">
    <h1 class=\\"display-5 \\">RSS aggregator</h1>
    <div id=\\"formConteiner\\">
        <form class=\\"form-inline\\" id=\\"rssAddForm\\">
            <input type=\\"text\\" name=\\"link\\" class=\\"form-control col mr-2\\" spellcheck=\\"false\\" placeholder=\\"RSS link\\">
            <button type=\\"submit\\" class=\\"btn btn btn-outline-primary d-flex align-items-center\\" disabled=\\"true\\">Add</button>
            <div class=\\"invalid-feedback\\" id=\\"feedback\\"></div>
        </form>
    </div>
</div>
<div class=\\"container-fluid\\">
    <div class=\\"row\\">
        <div class=\\"col col-3 p-0\\" id=\\"feeds\\">
            <div class=\\"list-group list-group-flush border-right\\">
                <div class=\\"list-group-item\\">
                    <h5 class=\\"mb-1\\">Lorem ipsum feed for an interval of 1 minutes</h5>
                    <p class=\\"mb-1\\">This is a constantly updating lorem ipsum feed</p>
                </div>
            </div>
        </div>
        <div class=\\"col col-8\\" id=\\"posts\\">
            <ul class=\\"list-group list-group-flush overflow-auto\\">
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637920\\">Veniam occaecat qui irure mollit ea ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637860\\">Do reprehenderit nostrud deserunt aliqua aliqua fugiat labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637800\\">Veniam magna qui mollit ea ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637740\\">Do nostrud laborum aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637680\\">Veniam magna occaecat qui irure mollit ea consequat ex pariatur.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637620\\">Do reprehenderit nostrud laborum deserunt aliqua aliqua fugiat magna labore ullamco.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637560\\">Qui amet et in pariatur et nostrud quis do id.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637500\\">Tempor magna non duis sint deserunt pariatur adipisicing elit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637440\\">Incididunt consequat reprehenderit sit adipisicing voluptate sint consequat.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637380\\">Labore aliqua incididunt duis voluptate reprehenderit officia est mollit voluptate velit consectetur nulla velit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637320\\">Voluptate deserunt elit consectetur tempor dolore ut dolor ex ullamco sunt aliquip esse sit.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637260\\">Eu adipisicing nostrud tempor voluptate commodo laboris eiusmod voluptate ex.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637200\\">Sint anim reprehenderit elit commodo.</a></li>
                <li class=\\"list-group-item\\"><a href=\\"http://example.com/test/1584637140\\">Ea officia veniam dolore adipisicing nulla esse aliqua.</a></li>
            </ul>
        </div>
    </div>
</div>"
`;
